### PR TITLE
Add doc, release and source build configs for Iron

### PR DESCRIPTION
**Pull request by Yadunund**
Follow up to #269 
Requires: https://github.com/ros/rosdistro/pull/36923

### CODE REVIEW

File: iron/release-build.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# ROS buildfarm release-build file
+---
+abi_incompatibility_assumed: true
+
+build_environment_variables:
+  RTI_NC_LICENSE_ACCEPTED: 'yes'
+
+jenkins_binary_job_priority: 80
+jenkins_binary_job_timeout: 120
+jenkins_source_job_priority: 70
```
Comment (cottsay) [inline iron/release-build.yaml]: These priorities should be updated to follow (foxy->galactic->humble)...

```suggestion
jenkins_binary_job_priority: 84
jenkins_binary_job_timeout: 120
jenkins_source_job_priority: 74
```

Same in arm64 and RHEL configs.

File: iron/source-build.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+# ROS buildfarm source-build file
+---
+build_environment_variables:
+  CONNEXTDDS_DIR: /opt/rti.com/rti_connext_dds-6.0.1
+  ROS_PYTHON_VERSION: 3
+  RTI_NC_LICENSE_ACCEPTED: 'yes'
+build_tool: colcon
+jenkins_commit_job_priority: 50
+jenkins_job_timeout: 120
+jenkins_pull_request_job_priority: 40
```
Comment (cottsay) [inline iron/source-build.yaml]:
```suggestion
jenkins_commit_job_priority: 54
jenkins_job_timeout: 120
jenkins_pull_request_job_priority: 44
```

File: iron/release-ubuntu-arm64-build.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+jenkins_source_job_timeout: 30
+notifications:
+  emails:
+  - steven+build.ros2.org@openrobotics.org
+  - ros2-buildfarm-iron@googlegroups.com
+  maintainers: true
+package_blacklist:
+- swri_console_util  # Build failures on all platforms. https://github.com/swri-robotics/marti_common/issues/665
+- swri_geometry_util  # Build failures on all platforms. https://github.com/swri-robotics/marti_common/issues/665
+- swri_roscpp  # Build failures on all platforms. https://github.com/swri-robotics/marti_common/issues/665
```
Comment (cottsay) [inline iron/release-ubuntu-arm64-build.yaml]: I believe that these problems were fixed. We should probably remove them from `rolling` at some point as well.

```suggestion
```

**Review by cottsay — CHANGES_REQUESTED** [review]
A couple of additional things:
1. Iron will be the first distro that doesn't run tests in system packaging jobs. This config needs to be added to each of the `release-*.yaml` configs:
   ```yaml
   package_dependency_behavior:
     include_test_dependencies: False
   ```
2. The new configs need to be referenced in the `index.yaml` at the root of this repo.

File: iron/release-ubuntu-arm64-build.yaml (hunk 3 above)
Comment (Yadunund) [reply iron/release-ubuntu-arm64-build.yaml]: Do i remove all the entires in `package_blacklist` or just the `swri` ones?

Comment (cottsay) [reply iron/release-ubuntu-arm64-build.yaml]: Just the `swri_*` ones - the RHEL `package_blacklist` should be fairly up-to-date.

File: iron/source-build.yaml (hunk 2 above)
Comment (Yadunund) [reply iron/source-build.yaml]: [177b305](https://github.com/ros2/ros_buildfarm_config/pull/270/commits/177b30569d4e03d849fe47a2f28caf022f109e62)

File: iron/release-build.yaml (hunk 1 above)
Comment (Yadunund) [reply iron/release-build.yaml]: [177b305](https://github.com/ros2/ros_buildfarm_config/pull/270/commits/177b30569d4e03d849fe47a2f28caf022f109e62)

File: iron/release-ubuntu-arm64-build.yaml (hunk 3 above)
Comment (Yadunund) [reply iron/release-ubuntu-arm64-build.yaml]: Sorry my comment was incoherent. Meant to ask if I should remove these entries from other `release-*` configs too. They seem to still be present in the RHEL one

**Review by clalancette — APPROVED** [review]
:tada:

**Comment by cottsay** [comment]
Current priorities dump:

<details>

```
$ python3 -m ros_buildfarm.scripts.misc.list_priorities index.yaml 
40
  jenkins_pull_request_job_priority::rolling/source-build.yaml
44
  jenkins_pull_request_job_priority::iron/source-build.yaml
45
  jenkins_pull_request_job_priority::humble/source-build.yaml
47
  jenkins_pull_request_job_priority::foxy/source-build.yaml
50
  jenkins_commit_job_priority::rolling/source-build.yaml
  jenkins_job_priority::rolling/ci-benchmark.yaml
  jenkins_job_priority::rolling/ci-clang-tidy.yaml
  jenkins_job_priority::rolling/ci-drake-ros-underlay.yaml
  jenkins_job_priority::rolling/ci-nightly-connext.yaml
  jenkins_job_priority::rolling/ci-nightly-cross-vendor-connext-cyclonedds.yaml
  jenkins_job_priority::rolling/ci-nightly-cross-vendor-connext-fastrtps-dynamic.yaml
  jenkins_job_priority::rolling/ci-nightly-cross-vendor-connext-fastrtps.yaml
  jenkins_job_priority::rolling/ci-nightly-cross-vendor-cyclonedds-fastrtps-dynamic.yaml
  jenkins_job_priority::rolling/ci-nightly-cross-vendor-cyclonedds-fastrtps.yaml
  jenkins_job_priority::rolling/ci-nightly-cross-vendor-fastrtps-fastrtps-dynamic.yaml
  jenkins_job_priority::rolling/ci-nightly-cyclonedds.yaml
  jenkins_job_priority::rolling/ci-nightly-debug.yaml
  jenkins_job_priority::rolling/ci-nightly-extra-rmw-release.yaml
  jenkins_job_priority::rolling/ci-nightly-fastrtps-dynamic.yaml
  jenkins_job_priority::rolling/ci-nightly-fastrtps.yaml
  jenkins_job_priority::rolling/ci-nightly-performance.yaml
  jenkins_job_priority::rolling/ci-nightly-release.yaml
  jenkins_job_priority::rolling/ci-overlay.yaml
54
  jenkins_commit_job_priority::iron/source-build.yaml
  jenkins_job_priority::iron/ci-benchmark.yaml
  jenkins_job_priority::iron/ci-nightly-connext.yaml
  jenkins_job_priority::iron/ci-nightly-cross-vendor-connext-cyclonedds.yaml
  jenkins_job_priority::iron/ci-nightly-cross-vendor-connext-fastrtps-dynamic.yaml
  jenkins_job_priority::iron/ci-nightly-cross-vendor-connext-fastrtps.yaml
  jenkins_job_priority::iron/ci-nightly-cross-vendor-cyclonedds-fastrtps-dynamic.yaml
  jenkins_job_priority::iron/ci-nightly-cross-vendor-cyclonedds-fastrtps.yaml
  jenkins_job_priority::iron/ci-nightly-cross-vendor-fastrtps-fastrtps-dynamic.yaml
  jenkins_job_priority::iron/ci-nightly-cyclonedds.yaml
  jenkins_job_priority::iron/ci-nightly-debug.yaml
  jenkins_job_priority::iron/ci-nightly-extra-rmw-release.yaml
  jenkins_job_priority::iron/ci-nightly-fastrtps-dynamic.yaml
  jenkins_job_priority::iron/ci-nightly-fastrtps.yaml
  jenkins_job_priority::iron/ci-nightly-performance.yaml
  jenkins_job_priority::iron/ci-nightly-release.yaml
  jenkins_job_priority::iron/ci-overlay.yaml
55
  jenkins_commit_job_priority::humble/source-build.yaml
  jenkins_job_priority::humble/ci-benchmark.yaml
  jenkins_job_priority::humble/ci-drake-ros-underlay.yaml
  jenkins_job_priority::humble/ci-nightly-connext.yaml
  jenkins_job_priority::humble/ci-nightly-cross-vendor-connext-cyclonedds.yaml
  jenkins_job_priority::humble/ci-nightly-cross-vendor-connext-fastrtps-dynamic.yaml
  jenkins_job_priority::humble/ci-nightly-cross-vendor-connext-fastrtps.yaml
  jenkins_job_priority::humble/ci-nightly-cross-vendor-cyclonedds-fastrtps-dynamic.yaml
  jenkins_job_priority::humble/ci-nightly-cross-vendor-cyclonedds-fastrtps.yaml
  jenkins_job_priority::humble/ci-nightly-cross-vendor-fastrtps-fastrtps-dynamic.yaml
  jenkins_job_priority::humble/ci-nightly-cyclonedds.yaml
  jenkins_job_priority::humble/ci-nightly-debug.yaml
  jenkins_job_priority::humble/ci-nightly-extra-rmw-release.yaml
  jenkins_job_priority::humble/ci-nightly-fastrtps-dynamic.yaml
  jenkins_job_priority::humble/ci-nightly-fastrtps.yaml
  jenkins_job_priority::humble/ci-nightly-performance.yaml
  jenkins_job_priority::humble/ci-nightly-release.yaml
  jenkins_job_priority::humble/ci-overlay.yaml
57
  jenkins_commit_job_priority::foxy/source-build.yaml
  jenkins_job_priority::foxy/ci-benchmark.yaml
  jenkins_job_priority::foxy/ci-nightly-connext.yaml
  jenkins_job_priority::foxy/ci-nightly-cross-vendor-connext-cyclonedds.yaml
  jenkins_job_priority::foxy/ci-nightly-cross-vendor-connext-fastrtps-dynamic.yaml
  jenkins_job_priority::foxy/ci-nightly-cross-vendor-connext-fastrtps.yaml
  jenkins_job_priority::foxy/ci-nightly-cross-vendor-cyclonedds-fastrtps-dynamic.yaml
  jenkins_job_priority::foxy/ci-nightly-cross-vendor-cyclonedds-fastrtps.yaml
  jenkins_job_priority::foxy/ci-nightly-cross-vendor-fastrtps-fastrtps-dynamic.yaml
  jenkins_job_priority::foxy/ci-nightly-cyclonedds.yaml
  jenkins_job_priority::foxy/ci-nightly-debug.yaml
  jenkins_job_priority::foxy/ci-nightly-extra-rmw-release.yaml
  jenkins_job_priority::foxy/ci-nightly-fastrtps-dynamic.yaml
  jenkins_job_priority::foxy/ci-nightly-fastrtps.yaml
  jenkins_job_priority::foxy/ci-nightly-performance.yaml
  jenkins_job_priority::foxy/ci-nightly-release.yaml
  jenkins_job_priority::foxy/ci-overlay.yaml
70
  jenkins_source_job_priority::rolling/release-build.yaml
  jenkins_source_job_priority::rolling/release-rhel-build.yaml
  jenkins_source_job_priority::rolling/release-ubuntu-arm64-build.yaml
74
  jenkins_source_job_priority::iron/release-build.yaml
  jenkins_source_job_priority::iron/release-rhel-build.yaml
  jenkins_source_job_priority::iron/release-ubuntu-arm64-build.yaml
75
  jenkins_source_job_priority::humble/release-build.yaml
  jenkins_source_job_priority::humble/release-rhel-build.yaml
  jenkins_source_job_priority::humble/release-ubuntu-arm64-build.yaml
77
  jenkins_source_job_priority::foxy/release-build.yaml
  jenkins_source_job_priority::foxy/release-focal-arm64-build.yaml
80
  jenkins_binary_job_priority::rolling/release-build.yaml
  jenkins_binary_job_priority::rolling/release-rhel-build.yaml
  jenkins_binary_job_priority::rolling/release-ubuntu-arm64-build.yaml
84
  jenkins_binary_job_priority::iron/release-build.yaml
  jenkins_binary_job_priority::iron/release-rhel-build.yaml
  jenkins_binary_job_priority::iron/release-ubuntu-arm64-build.yaml
85
  jenkins_binary_job_priority::humble/release-build.yaml
  jenkins_binary_job_priority::humble/release-rhel-build.yaml
  jenkins_binary_job_priority::humble/release-ubuntu-arm64-build.yaml
87
  jenkins_binary_job_priority::foxy/release-build.yaml
  jenkins_binary_job_priority::foxy/release-focal-arm64-build.yaml
89
  jenkins_job_priority::humble/doc-build.yaml
  jenkins_job_priority::iron/doc-build.yaml
  jenkins_job_priority::rolling/doc-build.yaml
```

</details>

**Review by cottsay — APPROVED** [review]
:building_construction:

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/preparing-for-iron-branching-on-rolling-2023-04-18/30980/5

**Pull request merged**